builder/js/ncf.js

```diff
@@ -46,232 +46,279 @@ app.directive('techniquename', function($filter) {
+  return angular.equals($scope.originalTechnique,technique);
+};
 
-  // Call ncf api to get techniques
-  $scope.getTechniques = function () {
+// Check if a method is selected
+$scope.isSelectedMethod = function(method) {
+  return angular.equals($scope.selectedMethod,method);
+};
 
-    $scope.techniques = [];
-    var data = {params: {path: $scope.path}}
-    $http.get('/ncf/api/techniques',data).
-      success(function(data, status, headers, config) {
+// Call ncf api to get techniques
+$scope.getTechniques = function () {
 
-        $.each( data.data.techniques, function(techniqueName, technique_raw) {
-          var technique = toTechUI(technique_raw);
-          $scope.techniques.push(technique);
-        })
+  $scope.techniques = [];
+  var data = {params: {path: $scope.path}}
+  $http.get('/ncf/api/techniques',data).
+    success(function(data, status, headers, config) {
 
-        // Display single errors
-        $.each( data.errors, function(index, error) {
-          errorNotification(error.message,error.details)
-        })
-      } ).
-      error($scope.handle_error(" while fetching techniques"));
-  };
+      $.each( data.data.techniques, function(techniqueName, technique_raw) {
+        var technique = toTechUI(technique_raw);
+        $scope.techniques.push(technique);
+      })
 
-  // Call ncf api to get genereric methods and then after that get Techniques
-  $scope.getMethodsAndTechniques = function () {
-    var data = {params: {path: $scope.path}}
-    $http.get('/ncf/api/generic_methods', data).
-      success(function(data, status, headers, config) {
-        $scope.generic_methods = data.data.generic_methods;
-        $scope.methodsByCategory = $scope.groupMethodsByCategory();
-        $scope.authenticated = true;
-        // Once we have our methods we can fetch our techniques which depends on them
-        $scope.getTechniques();
-
-        // Display single errors
-        $.each( data.errors, function(index, error) {
-          errorNotification(error.message,error.details)
-        })
-      } ).
-      error($scope.handle_error(" while fetching generic methods"));
-  };
+      // Display single errors
+      $.each( data.errors, function(index, error) {
+        errorNotification(error.message,error.details)
+      })
+    } ).
+    error($scope.handle_error(" while fetching techniques"));
+};
+
+// Call ncf api to get genereric methods and then after that get Techniques
+$scope.getMethodsAndTechniques = function () {
+  var data = {params: {path: $scope.path}}
+  $http.get('/ncf/api/generic_methods', data).
+    success(function(data, status, headers, config) {
+      $scope.generic_methods = data.data.generic_methods;
+      $scope.constraints = data.data.constraints;
+      $scope.methodsByCategory = $scope.groupMethodsByCategory();
+      $scope.authenticated = true;
+      // Once we have our methods we can fetch our techniques which depends on them
+      $scope.getTechniques();
```

Comment:
This fails if not techniques is defined (must data.data.techniques !== undefined in gettechniques, or something alike)